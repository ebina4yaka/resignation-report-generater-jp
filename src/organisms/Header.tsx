import { ReactElement } from 'react'
import Typography from '@material-ui/core/Typography'
import Toolbar from '@material-ui/core/Toolbar'
import AppBar from '@material-ui/core/AppBar'
import { createStyles, Theme } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import DescriptionIcon from '@material-ui/icons/Description'

type Props = {
  title: string
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    icon: {
      marginRight: theme.spacing(2),
      marginTop: theme.spacing(0.4),
    },
    title: {
      flexGrow: 1,
    },
  })
)

export default function Header(props: Props): ReactElement {
  const { title } = props
  const classes = useStyles()
  return (
    <div className={classes.root}>
      <AppBar position="relative">
        <Toolbar>
          <div className={classes.icon}>
            <DescriptionIcon />
          </div>
          <Typography variant="h6" color="inherit" className={classes.title}>
            {title}
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  )
}
