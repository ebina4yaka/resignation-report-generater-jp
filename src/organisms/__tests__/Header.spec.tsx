import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../Header'

test('Snapshot test for Header', () => {
  const tree = renderer.create(<Header headerTitle="test" />).toJSON()
  expect(tree).toMatchSnapshot()
})
