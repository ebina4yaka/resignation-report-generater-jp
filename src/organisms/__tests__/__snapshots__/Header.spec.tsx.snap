// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Header 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    />
  </header>
</div>
`;
